integration_test/gce/gce_testing.go

```diff
@@ -739,6 +739,95 @@ func runCommand(ctx context.Context, logger *log.Logger, stdin io.Reader, args [
 	return output, err
 }
 
+// getGcloudConfigDir returns the current gcloud configuration directory.
+func getGcloudConfigDir(ctx context.Context) (string, error) {
+	null, err := os.Open(os.DevNull)
+	if err != nil {
+		panic(fmt.Sprintf("unable to open os.DevNull: %v", err))
+	}
+	out, err := RunGcloud(ctx, log.New(null, "", 0), "", []string{"info", "--format=value[terminator=''](config.paths.global_config_dir)"})
+	if err != nil {
+		return "", fmt.Errorf("error running gcloud info: %w", err)
+	}
+	return out.Stdout, nil
+}
+
+// copyFile copies the contents of source into target.
+func copyFile(source string, target string) error {
+	in, err := os.Open(source)
+	if err != nil {
+		return err
+	}
+	defer in.Close()
+	out, err := os.Create(target)
+	if err != nil {
+		return err
+	}
+	_, copyErr := io.Copy(out, in)
+	syncErr := out.Sync()
+	closeErr := out.Close()
+	return multierr.Combine(copyErr, syncErr, closeErr)
+}
+
+// SetupGcloudConfigDir sets up a new gcloud configuration directory.
+// This copies the "configurations" subdirectory of the context-specified
+// configuration directory into the new directory.
+func SetupGcloudConfigDir(ctx context.Context, directory string) error {
+	relevantFiles := []string{
+		"active_config",
+		"access_tokens.db",
+		"credentials.db",
+		"default_configs.db",
+	}
+	currentConfigDir, err := getGcloudConfigDir(ctx)
+	if err != nil {
+		return err
+	}
+	sfi, err := os.Stat(currentConfigDir)
```

Comment:
this is a lot of code! [Stackoverflow](https://stackoverflow.com/a/77849003) says that go 1.23 will likely get a function to do (approximately) this, but in the meantime, why not just runCommand() a "cp --recursive" command? Do we really need to copy only these relevantFiles or something?

Reply:
I think I was proposing:
- cp -r for now, with a TODO I guess
- replace that with os.CopyFS when it becomes available, because it's less hacky

This saves us about 50+ lines of code of hand-rolled directory copying code. Like for example, `defer in.Close()` drops errors from closing files, but we don't need to hash that out if you are just calling `cp -r`. So yeah I think `cp -r` is worth doing, even if only to save time going back and forth in reviews.

Reply:
Done.